frontend/src/components/common/NumberSelector/index.tsx

```diff
@@ -11,16 +11,19 @@ interface NumberSelectorProps extends TextInputProps {
     fieldName: string,
     min?: number;
     max?: number;
+    sharedValues?: SharedValues;
 }
 
-export const NumberSelector = ({formInstance, fieldName, min, max}: NumberSelectorProps) => {
+export const NumberSelector = ({formInstance, fieldName, min, max, sharedValues}: NumberSelectorProps) => {
     const handlers = useRef<NumberInputHandlers>(null);
     // Start with 0, ensuring it's treated as number for consistency
     const [value, setValue] = useState<number>(0);
 
     const minValue = min || 0;
     const maxValue = max || 100;
 
+    const [sharedVals, setSharedVals] = useState<SharedValues>(sharedValues ?? new SharedValues(maxValue));
```

Comment:
You should be able to avoid using `useState` here, as `setSharedVals` is never used.
```suggestion
    	const sharedVals = sharedValues ?? new SharedValues(maxValue);
```

Reply:
I had something like this but I was having trouble getting it to work properly... I'm new with React, so I was probably doing something wrong. I'll address this, thanks!

Reply:
Yeah, I checked again and when I don't use `useState`, the `sharedVals` property doesn't get properly updated. The only way I seem able to get the state shared between the components is via `useState`.